modules/statistics/statistics.js

```diff
@@ -230,34 +247,89 @@ Statistics.prototype.stopRemoteStats = function() {
 
 /**
  * Initializes the callstats.io API.
- * @param peerConnection {JingleSessionPC} the session object
+ * @param {TraceablePeerConnection} tpc the {@link TraceablePeerConnection}
+ * instance for which CalStats will be started.
+ * @param {string} remoteUserID
  */
-Statistics.prototype.startCallStats = function(session) {
-    if (this.callStatsIntegrationEnabled && !this.callStatsStarted) {
-        // Here we overwrite the previous instance, but it must be bound to
-        // the new PeerConnection
-        this.callstats = new CallStats(session, this.options);
-        Statistics.callsStatsInstances.push(this.callstats);
-        this.callStatsStarted = true;
+Statistics.prototype.startCallStats = function(tpc, remoteUserID) {
+    if (!this.callStatsIntegrationEnabled) {
+        return;
+    } else if (this.callsStatsInstances.has(tpc.id)) {
+        logger.error('CallStats instance for ${tpc} exists already');
+
+        return;
+    }
+
+    if (!CallStats.isBackendInitialized()) {
```

Comment:
So, I'm thinking, this should be moved into CallStats itself. As in, we pass all the options to the constructor, and if there is no backend initialized it will do it on its own internally, that way the backend is "hidden" from the statistics module. WDYT?